Library/Homebrew/cmd/update-report.rb

```diff
@@ -101,6 +104,32 @@ def install_core_tap_if_necessary
     ENV["HOMEBREW_UPDATE_BEFORE_HOMEBREW_HOMEBREW_CORE"] = revision
     ENV["HOMEBREW_UPDATE_AFTER_HOMEBREW_HOMEBREW_CORE"] = revision
   end
+
+  def migrate_legacy_cache_if_necessary
+    legacy_cache = Pathname.new "/Library/Caches/Homebrew"
+    return if HOMEBREW_CACHE.to_s == legacy_cache.to_s
+    return unless legacy_cache.directory?
+    return unless legacy_cache.readable_real?
+
+    # Cleanup to avoid copying files unnecessarily
+    ohai "Cleaning up #{legacy_cache}..."
+    Cleanup.cleanup_cache legacy_cache
+
+    # This directory could have been compromised if it's world-writable/
+    # a symlink/owned by another user so don't copy files in those cases.
+    return if legacy_cache.world_writable?
+    return if legacy_cache.symlink?
+    return if !legacy_cache.owned? && legacy_cache.lstat.uid != 0
+
+    ohai "Migrating #{legacy_cache} to #{HOMEBREW_CACHE}..."
+    HOMEBREW_CACHE.mkpath
+    legacy_cache.entries.each {|f| FileUtils.cp_r f, HOMEBREW_CACHE }
```

Comment:
I'm thinking maybe start over clean and treat it as a documentation issue, with a notice pointing to a support article explaining how to move the directory with Finder and/or Terminal if desired, and leave it at the users' and adminstrators' discretion. That also has the benefit of leaving the original /Library/Caches/Homebrew unmolested in case someone's desire (not mine to be clear – I switched to ~/Library/Caches/Homebrew a long time ago) is to just set `export HOMEBREW_CACHE=/Library/Caches/Homebrew` and move on.